ragaai_catalyst/tracers/agentic_tracing/utils/llm_utils.py

```
@@ -604,3 +602,18 @@ def extract_llm_data(args, kwargs, result):
+    # Use tiktoken to count tokens
+    try:
+        import tiktoken
+        
+        # Use GPT-4o model's encoding (cl100k_base)
+        encoding = tiktoken.get_encoding("cl100k_base")
+        
+        # Count tokens
+        tokens = encoding.encode(input_str)
+        return len(tokens)
+    except Exception:
+        raise Exception("Failed to count tokens")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)